projects/js-packages/ai-client/src/components/ai-control/ai-control.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { PlainText } from '@wordpress/block-editor';
+import classNames from 'classnames';
+import React from 'react';
+/**
+ * Internal dependencies
+ */
+import AiStatusIndicator from '../ai-status-indicator/index.js';
+import './style.scss';
+/**
+ * Types
+ */
+import type { RequestingStateProp } from '../../types.js';
+import type { ReactElement } from 'react';
+
+type AIControlProps = {
+	disabled?: boolean;
+	value: string;
+	placeholder?: string;
+	isTransparent?: boolean;
+	state?: RequestingStateProp;
+	onChange?: ( newValue: string ) => void;
+	bannerComponent?: ReactElement;
+	errorComponent?: ReactElement;
+	actions?: ReactElement;
+	message?: ReactElement;
+	promptUserInputRef?: React.MutableRefObject< HTMLInputElement >;
+};
+
+/**
+ * Base AIControl component. Contains the main structure of the control component and slots for banner, error, actions and message.
+ *
+ * @param {AIControlProps} props - Component props
+ * @returns {ReactElement}       Rendered component
+ */
+export default function AIControl( {
+	disabled = false,
+	value = '',
+	placeholder = '',
+	isTransparent = false,
+	state = 'init',
+	onChange,
+	bannerComponent = null,
+	errorComponent = null,
```

Comment:
A complete nitpick, but sounds strange for me have the `Component` affix, since we don't have in `actions` and `message`.

Reply:
Changing to `banner` and `error`. Though those will eventually be removed, as the `message` slot should do the work of both. They are kept for now for retrocompatibility purposes.